qiskit/extensions/hamiltonian_gate.py

```diff
@@ -127,7 +130,25 @@ def validate_parameter(self, parameter):
 
 
 def hamiltonian(self, operator, time, qubits, label=None):
-    """Apply hamiltonian evolution to qubits."""
+    """Apply hamiltonian evolution to qubits.
+
+    This gate resolves to a :class:`~qiskit.extensions.UnitaryGate` as
+    :math:`U(t) = exp(-1j * t * H)`, which can be decomposed into basis gates
+    if it is 2 qubits or less, or simulated directly in Aer for more qubits.
+
+    Args:
+        operator (matrix or Operator): a hermitian operator.
+        time (float): time evolution parameter.
+        qubits (Union[int, Tuple[int]]): The circuit qubits to apply the
+            transformation to.
+        label (str): unitary name for backend [Default: None].
```

Comment:
Could you put the typehints in the function definition instead of the docs here? 🙂

Reply:
Almost none of the other methods here or in `unitary.py` have type hints though. Moving these to type hints would make it quite inconsistent, and I think it'd be better as a candidate for a separate PR, don't you think?

Reply:
Ok makes sense 👍🏻